packages/front-end/components/SlackIntegrations/SlackIntegrationsListView/SlackIntegrationsListView.tsx

```diff
@@ -0,0 +1,297 @@
+        data,
+        id,
+      });
+    },
+    []
+  );
+
+  const handleOnCreateModalOpen = useCallback(() => {
+    setModalMode({
+      mode: "create",
+    });
+  }, []);
+
+  const [addEditError, setAddEditError] = useState<null | string>(null);
+
+  const { data, mutate, error: loadError } = useApi<{
+    slackIntegrations: SlackIntegrationInterface[];
+  }>("/integrations/slack");
+
+  const errorMessage = loadError?.message || null;
+
+  const slackIntegrations = data?.slackIntegrations || [];
+
+  const handleDelete = useCallback(
+    async (id: string) => {
+      await apiCall<{
+        error?: string;
+        slackIntegration?: SlackIntegrationInterface;
+      }>(`/integrations/slack/${id}`, {
+        method: "DELETE",
+      });
+
+      await mutate();
+    },
+    [apiCall, mutate]
+  );
+
+  const handleCreate = useCallback(
+    async (data: SlackIntegrationEditParams) => {
+      setAddEditError(null);
+
+      try {
+        const response = await apiCall<{
+          error?: string;
+          slackIntegration?: SlackIntegrationInterface;
+        }>("/integrations/slack", {
+          method: "POST",
+          body: JSON.stringify(data),
+        });
+
+        if (response.error) {
+          setAddEditError(
+            `Failed to create Slack integration: ${
+              response.error || "Unknown error"
+            }`
+          );
+        } else {
+          setAddEditError(null);
+          setModalMode(null);
+          mutate();
+        }
+      } catch (e) {
+        setAddEditError(`Failed to create Slack integration: ${e.message}`);
+      }
+    },
+    [apiCall, mutate]
+  );
+
+  const handleUpdate = useCallback(
+    async (id: string, data: SlackIntegrationEditParams) => {
+      setAddEditError(null);
+
+      try {
+        const response = await apiCall<{
+          error?: string;
+          slackIntegration?: SlackIntegrationInterface;
+        }>(`/integrations/slack/${id}`, {
+          method: "PUT",
+          body: JSON.stringify(
+            pick(data, [
+              "name",
+              "description",
+              "projects",
+              "environments",
+              "events",
+              "tags",
+              "slackAppId",
+              "slackSigningKey",
+              "slackIncomingWebHook",
+            ])
+          ),
+        });
+
+        if (response.error) {
+          setAddEditError(
+            `Failed to update Slack integration: ${
+              response.error || "Unknown error"
+            }`
+          );
+        } else {
+          setAddEditError(null);
+          setModalMode(null);
+          mutate();
+        }
+      } catch (e) {
+        setAddEditError(`Failed to update Slack integration: ${e.message}`);
+      }
+    },
+    [apiCall, mutate]
+  );
+
+  const environmentSettings = useEnvironments();
+  const environments = environmentSettings.map((env) => env.id);
+
+  const { projects, tags } = useDefinitions();
```

Comment:
You could move environments, projects, and tags into the `SlackIntegrationsListView` component if you want.  Looks like they aren't used for anything here besides being passed through.

Reply:
ok, makes sense